### PR TITLE
[JSC] Rename Array#groupBy to Array#group and enable them

### PR DESCRIPTION
**Pull request by Constellation**
#### 317e70035de4f830152ae3fc5ed483f10c837f5d
<pre>
[JSC] Rename Array#groupBy to Array#group and enable them
<a href="https://bugs.webkit.org/show_bug.cgi?id=243510">https://bugs.webkit.org/show_bug.cgi?id=243510</a>

Reviewed by Mark Lam, Justin Michaud, Devin Rousso and Ross Kirsling.

We agreed at TC39 to renaming Array#groupBy and Array#groupByToMap to Array#group and Array#groupToMap
to keep web-compatibility. This patch renames and enables them, stage-4 progress is tracked in [1].
We also upstreamed test262 fix for this[2].

[1]: <a href="https://github.com/tc39/proposal-array-grouping/issues/32">https://github.com/tc39/proposal-array-grouping/issues/32</a>
[2]: <a href="https://github.com/tc39/test262/pull/3632">https://github.com/tc39/test262/pull/3632</a>

* Source/JavaScriptCore/builtins/ArrayPrototype.js:
(group):
(groupToMap):
(groupBy): Deleted.
(groupByToMap): Deleted.
* Source/JavaScriptCore/runtime/ArrayPrototype.cpp:
(JSC::ArrayPrototype::finishCreation):
* Source/JavaScriptCore/runtime/OptionsList.h:

Canonical link: <a href="https://commits.webkit.org/253101@main">https://commits.webkit.org/253101@main</a>
</pre>

### CODE REVIEW

**Review by rkirsling — APPROVED** [review]
:shipit:

File: Source/JavaScriptCore/runtime/OptionsList.h [hunk 1]
```diff
@@ -541,7 +541,7 @@ bool canUseWebAssemblyFastMemory();
     /* Feature Flags */\
     \
     v(Bool, useArrayFindLastMethod, true, Normal, "Expose the findLast() and findLastIndex() methods on Array and %TypedArray%.") \
-    v(Bool, useArrayGroupByMethod, false, Normal, "Expose the groupBy() and groupByToMap() methods on Array.") \
+    v(Bool, useArrayGroupMethod, true, Normal, "Expose the group() and groupToMap() methods on Array.") \
```
Comment (justinmichaud) [inline Source/JavaScriptCore/runtime/OptionsList.h:544]: We enable this by default?

**Review by MenloDorian — APPROVED** [review]
r=me too

**Review by justinmichaud — APPROVED** [review]
r=me with question

File: Source/JavaScriptCore/runtime/OptionsList.h (hunk 1 above)
Comment (Constellation) [reply Source/JavaScriptCore/runtime/OptionsList.h:544]: Yes, this patch enables it.

**Review by dcrousso — APPROVED** [review]
can you also update the entries in `WI.NativePrototypeFunctionParameters` for Web Inspector? :)

**Pull request merged**